protostar/compiler/compiled_contract_writer.py

```diff
@@ -0,0 +1,45 @@
+import json
+from pathlib import Path
+
+from starkware.starknet.services.api.contract_class import ContractClass
+
+
+class CompiledContractWriter:
+    def __init__(self, contract: ContractClass, contract_name: str) -> None:
+        self._contract = contract
+        self._contract_name = contract_name
+
+    def save(self, output_dir: Path) -> None:
+        self.save_compiled_contract(output_dir)
+        self.save_compiled_contract_abi(output_dir)
+
+    def save_compiled_contract(self, output_dir: Path) -> None:
+        self._create_output_dir(output_dir)
+        with open(
+            Path(output_dir, f"{self._contract_name}.json"), mode="w", encoding="utf-8"
+        ) as output_file:
+            json.dump(
+                self._contract.Schema().dump(self._contract),
+                output_file,
+                indent=4,
+                sort_keys=True,
+            )
+            output_file.write("\n")
```

Comment:
the pattern of opening a file X and writing Y into it as pretty formatted JSON with trailing whitespace is duplicated here